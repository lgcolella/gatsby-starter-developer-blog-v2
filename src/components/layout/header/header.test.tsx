/* Vendor imports */
import React from 'react'
import { shallow } from 'enzyme'
/* App imports */
import Header from './index'

describe('Header component', () => {
  beforeAll(() => {
    jest.useFakeTimers()
  })

  test('matches snapshot', () => {
    let wrapper = shallow(<Header />)

    expect(wrapper).toMatchSnapshot()
  })

  test('matches snapshot whether it is fixed or not', () => {
    let wrapper = shallow(<Header />)

    wrapper.setState({ fixedHeader: true })
    expect(wrapper).toMatchSnapshot()

    wrapper.setState({ fixedHeader: false })
    expect(wrapper).toMatchSnapshot()
  })

  test('toggles the `fixed` state on scroll event', () => {
    let eventHandler: any = {}
    window.addEventListener = jest.fn((event, cb) => {
      eventHandler[event] = cb
    })
    const wrapper = shallow(<Header />)

    // case #1: header is not fixed if scrollY < 100
    // @ts-ignore
    window.scrollY = 50
    eventHandler.scroll() // trigger event
    jest.runAllTimers() // internally `setTimeout` is used to throttle the event
    expect(wrapper.state('lastScrollY')).toBe(50)
    expect(wrapper.state('fixedHeader')).toBe(false)

    // case #2: header is fixed if scrollY > 100
    // @ts-ignore
    window.scrollY = 102
    eventHandler.scroll() // Trigger event
    jest.runAllTimers() // internally `setTimeout` is used to throttle the event
    expect(wrapper.state('lastScrollY')).toBe(102)
    expect(wrapper.state('fixedHeader')).toBe(true)

    // case #3: header is not fixed if scrollY > 'previous' scrollY
    // @ts-ignore
    window.scrollY = 101
    eventHandler.scroll() // Trigger event
    jest.runAllTimers() // internally `setTimeout` is used to throttle the event
    expect(wrapper.state('lastScrollY')).toBe(101)
    expect(wrapper.state('fixedHeader')).toBe(false)
  })
})
