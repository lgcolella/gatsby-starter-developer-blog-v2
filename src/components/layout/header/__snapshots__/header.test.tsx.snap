// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div
  className="container visibleContainer"
>
  <div
    className="innerContainer"
  >
    <div
      className="title"
    >
      <mockConstructor
        to="/"
      >
        <h3>
          Vincent van Gogh
        </h3>
      </mockConstructor>
      <button>
        <FaBars
          onClick={[Function]}
          size="30"
        />
      </button>
    </div>
    <div
      className="pages collapsedMenu"
    >
      <ul>
        <li>
          <mockConstructor
            to="/blog/"
          >
            Blog
          </mockConstructor>
        </li>
        <li>
          <mockConstructor
            to="/tag/"
          >
            Tags
          </mockConstructor>
        </li>
        <li>
          <mockConstructor
            to="/about/"
          >
            About
          </mockConstructor>
        </li>
      </ul>
    </div>
    <div
      className="social collapsedMenu"
    >
      <ul>
        <li>
          <a
            href="https://github.com/lgcolella/gatsby-starter-developer-blog-v2"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <FaGithub
              size="30"
            />
          </a>
        </li>
        <li>
          <a
            href="#"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <FaLinkedin
              size="30"
            />
          </a>
        </li>
        <li>
          <a
            href="#"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <FaRss
              size="30"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Header component matches snapshot whether it is fixed or not 1`] = `
<div
  className="container hiddenContainer"
>
  <div
    className="innerContainer"
  >
    <div
      className="title"
    >
      <mockConstructor
        to="/"
      >
        <h3>
          Vincent van Gogh
        </h3>
      </mockConstructor>
      <button>
        <FaBars
          onClick={[Function]}
          size="30"
        />
      </button>
    </div>
    <div
      className="pages collapsedMenu"
    >
      <ul>
        <li>
          <mockConstructor
            to="/blog/"
          >
            Blog
          </mockConstructor>
        </li>
        <li>
          <mockConstructor
            to="/tag/"
          >
            Tags
          </mockConstructor>
        </li>
        <li>
          <mockConstructor
            to="/about/"
          >
            About
          </mockConstructor>
        </li>
      </ul>
    </div>
    <div
      className="social collapsedMenu"
    >
      <ul>
        <li>
          <a
            href="https://github.com/lgcolella/gatsby-starter-developer-blog-v2"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <FaGithub
              size="30"
            />
          </a>
        </li>
        <li>
          <a
            href="#"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <FaLinkedin
              size="30"
            />
          </a>
        </li>
        <li>
          <a
            href="#"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <FaRss
              size="30"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Header component matches snapshot whether it is fixed or not 2`] = `
<div
  className="container visibleContainer"
>
  <div
    className="innerContainer"
  >
    <div
      className="title"
    >
      <mockConstructor
        to="/"
      >
        <h3>
          Vincent van Gogh
        </h3>
      </mockConstructor>
      <button>
        <FaBars
          onClick={[Function]}
          size="30"
        />
      </button>
    </div>
    <div
      className="pages collapsedMenu"
    >
      <ul>
        <li>
          <mockConstructor
            to="/blog/"
          >
            Blog
          </mockConstructor>
        </li>
        <li>
          <mockConstructor
            to="/tag/"
          >
            Tags
          </mockConstructor>
        </li>
        <li>
          <mockConstructor
            to="/about/"
          >
            About
          </mockConstructor>
        </li>
      </ul>
    </div>
    <div
      className="social collapsedMenu"
    >
      <ul>
        <li>
          <a
            href="https://github.com/lgcolella/gatsby-starter-developer-blog-v2"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <FaGithub
              size="30"
            />
          </a>
        </li>
        <li>
          <a
            href="#"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <FaLinkedin
              size="30"
            />
          </a>
        </li>
        <li>
          <a
            href="#"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <FaRss
              size="30"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
